### PR TITLE
Implementation of basic socket.io connection - fixes #5

### LINKED ISSUE #5: Implement basic socket.io connection

### PR DESCRIPTION
**Pull request by azixus**
Basic socket.io server - client connection with tests.

### CODE REVIEW

File: src/debatemanager.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        this.io.on('connect', (socket) => {
+            logger.debug(`New socket (${socket.id}) connected to server`);
+
+            socket.on('disconnect', (reason) => {
+                logger.debug(`Socket (${socket.id}) disconnected`);
+            });
+        });
+    }
+
+    /**
```
Comment (ghost) [inline src/debatemanager.js:58]: The IO socket server is not stopped. Is it wanted ?

File: test/server.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            done();
+        });
+    });
+
+    describe('socket.io connection test', () => {
+        let client;
+
+        beforeEach(() => {
+            client = io.connect(srvAddress, {
+                path: '/socket-io',
```
Comment (ghost) [inline test/server.js]: You can use `static DEFAULT_PATH = '/socket-io';` from SocketConfig class.
```suggestion
                path: '${SocketConfig.DEFAULT_PATH}',
```

**Review by ghost — CHANGES_REQUESTED** [review]
This seems to be a very good basis for the server implementation. Nice job!
I just need your responses from a question and a suggestion before accepting your PL.

File: src/debatemanager.js [hunk 3]
```diff
@@ -56,6 +57,12 @@ export class DebateManager {
...
     stop() {
         logger.info("Server stopping");
-        this.webServer.close();
+        this.io.close();
+
+        // Use http-terminator to gracefully terminate the server
+        const httpTerminator = http_terminator.createHttpTerminator({
+            server: this.webServer
+        });
+        httpTerminator.terminate();
```
Comment (AziXus) [inline src/debatemanager.js]: Socket server is now closed and http server gracefully terminated.

File: src/debatemanager.js (hunk 1 above)
Comment (azixus) [reply src/debatemanager.js:58]: The socket.io server is based on the webServer. Closing the webServer automatically stops the socket.io server.

**Pull request merged**